apps/simple-trading-app/src/app/hooks/use-order-margin.ts

```diff
@@ -66,41 +66,30 @@ const types: Record<VegaWalletOrderType, OrderType> = {
 
 const useOrderMargin = ({ order, market, partyId }: Props) => {
   const marketPositions = useMarketPositions({ marketId: market.id, partyId });
-  console.log('marketPositions', marketPositions);
-  console.log(
-    'marketPositions.openVolume',
-    marketPositions?.openVolume.toNumber()
-  );
-  console.log(
-    'marketPositions.balanceSum',
-    marketPositions?.balanceSum.toNumber()
-  );
   const { data } = useQuery<EstimateOrder, EstimateOrderVariables>(
     ESTIMATE_ORDER_QUERY,
     {
       variables: {
         marketId: market.id,
         partyId,
         price: market.depth.lastTrade?.price,
-        size: order.size + (marketPositions?.openVolume.toNumber() || 0),
+        size: new BigNumber(marketPositions?.openVolume || 0)
+          .plus(order.size)
+          .toString(),
         side: order.side === VegaWalletOrderSide.Buy ? Side.Buy : Side.Sell,
         timeInForce: times[order.timeInForce],
         type: types[order.type],
       },
-      skip: !partyId || !market.id || !order.size,
+      skip:
+        !partyId || !market.id || !order.size || !market.depth.lastTrade?.price,
     }
   );
   if (data?.estimateOrder.marginLevels.initialLevel) {
-    return formatNumber(
+    return addDecimal(
       Math.max(
         0,
-        new BigNumber(
-          addDecimal(
-            data.estimateOrder.marginLevels.initialLevel,
-            market.decimalPlaces
-          )
-        )
-          .minus(marketPositions?.balanceSum.toNumber() || 0)
+        new BigNumber(data.estimateOrder.marginLevels.initialLevel)
+          .minus(marketPositions?.balance || 0)
```

Comment:
it should be toString, toNumber reach Number.MAX_SAFE_INTEGER

Reply:
done